### PR TITLE
Remove deprecated functions

### PR DESCRIPTION
**Pull request by kylebarron**
Regarding removing `rio_tiler.mosaic.mosaic_tiler`: I feel that if you’re doing a major version bump anyways, there’s no reason to re-export a function with two names.

In any case, `mosaic_tiler` isn't currently backwards compatible because it returns the output of `mosaic_reader` directly, but `mosaic_reader` was updated to return the `asset_ids` of the assets that were used. If you decide to keep `mosaic_tiler` for backwards compatibility with `rio_tiler_mosaic`, it would be good to return only the `tile, mask`.

### CODE REVIEW

File: rio_tiler/mosaic/reader.py [hunk 1]
```diff
@@ -87,28 +87,3 @@ def reader(asset: str, *args, **kwargs) -> Tuple[numpy.ndarray, numpy.ndarray]:
...
-        assets,
-        reader,
-        tile_x,
-        tile_y,
-        tile_z,
-        pixel_selection=pixel_selection,
-        chunk_size=chunk_size,
-        threads=threads,
-        **kwargs,
-    )
```
Comment (vincentsarago) [inline rio_tiler/mosaic/reader.py:114]: 👍

File: rio_tiler/reader.py [hunk 2]
```diff
@@ -247,13 +246,6 @@ def part(
             height=orig_vrt_height,
         )
 
-    if warp_vrt_option:
-        warnings.warn(
-            "warp_vrt_option will be removed in 2.0, use vrt_options",
-            DeprecationWarning,
-        )
-        vrt_options = vrt_options or warp_vrt_option
-
```
Comment (vincentsarago) [inline rio_tiler/reader.py:256]: I think you didn't remove the options in the function definition!

Comment (kylebarron) [reply rio_tiler/reader.py:256]: oh yeah you're right

**Comment by vincentsarago** [comment]
@kylebarron i'll let you fix the merge conflict and merge

**Pull request merged**